smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/TypeScriptSettings.java

```diff
@@ -295,6 +304,28 @@ public void setDisableDefaultValidation(boolean disableDefaultValidation) {
         this.disableDefaultValidation = disableDefaultValidation;
     }
 
+    /**
+     * Returns the compatibility mode in regards to required members..
+     *
+     * @return the configured compatibility mode in regards to required members..
+     * Defaults to {@link RequiredMemberMode#NULLABLE}
+     */
+    public RequiredMemberMode getRequiredMemberMode() {
+        return requiredMemberMode;
+    }
+
+    public void setRequiredMemberMode(
+        RequiredMemberMode requiredMemberMode) {
+            if (requiredMemberMode != RequiredMemberMode.NULLABLE) {
+                LOGGER.warning(String.format("By setting the member nullability compatibility mode to '%s', a"
+                    + " member that has the '@required' trait applied CANNOT be 'undefined'. Differrent than"
+                    + " when it is set to '%s', it will be considered a BACKWARDS INCOMPATIBLE change for"
```

Comment:
I think the "Different than when it is set to NULLABLE" is making it bit confusing to read and not useful IMO. I'd just drop that part.

```suggestion
                    + " member that has the '@required' trait applied CANNOT be 'undefined'."
                    + " It will be considered a BACKWARDS INCOMPATIBLE change for"
```
with reformatting.